robotx_behavior_tree/plugins/action/move_to_gate_action.cpp

```diff
@@ -0,0 +1,241 @@
+#include "tf2_ros/buffer.h"
+#include "tf2_ros/transform_broadcaster.h"
+#include "tf2_ros/transform_listener.h"
+
+namespace robotx_behavior_tree
+{
+class MoveToGateAction : public ActionROS2Node
+{
+public:
+  MoveToGateAction(const std::string & name, const BT::NodeConfiguration & config)
+  : ActionROS2Node(name, config), buffer_(get_clock()), listener_(buffer_)
+  {
+    declare_parameter("goal_tolerance", 1.0);
+    get_parameter("goal_tolerance", goal_tolerance_);
+    goal_pub_gate_ =
+      this->create_publisher<geometry_msgs::msg::PoseStamped>("/move_base_simple/goal", 1);
+  }
+
+private:
+  rclcpp::TimerBase::SharedPtr update_position_timer_;
+  float red_buoy_x = 0.0;
+  float red_buoy_y = 0.0;
+  float green_buoy_x = 0.0;
+  float green_buoy_y = 0.0;
+  float goal_x;
+  float goal_y;
+  float goal_theta;
+
+  float distance_;
+  double goal_tolerance_;
+
+  tf2_ros::Buffer buffer_;
+  tf2_ros::TransformListener listener_;
+  rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr goal_pub_gate_;
+  geometry_msgs::msg::PoseStamped goal_;
+
+protected:
+  BT::NodeStatus onStart() override
+  {
+    std::vector<robotx_behavior_msgs::msg::TaskObject> green_buoys_array;
+    std::vector<robotx_behavior_msgs::msg::TaskObject> red_buoys_array;
+    const auto status_planner = getPlannerStatus();
+
+    if (status_planner) {
+      RCLCPP_INFO(get_logger(), "status : %d", status_planner.value()->status);
+    }
+
+    try {
+      const auto task_objects_array = getTaskObjects();
+      if (task_objects_array) {
+        RCLCPP_INFO(
+          get_logger(), "task object size : %ld", task_objects_array->task_objects.size());
+      }
+
+      // return BT::NodeStatus::FAILURE;
+      for (size_t i = 0; i < task_objects_array->task_objects.size(); i++) {
+        RCLCPP_INFO(
+          get_logger(), "task_object_kind : %d", task_objects_array->task_objects[i].object_kind);
+        RCLCPP_INFO(get_logger(), "task_object_x : [%f]", task_objects_array->task_objects[i].x);
+        RCLCPP_INFO(get_logger(), "task_object_y : [%f]", task_objects_array->task_objects[i].y);
+        if (task_objects_array->task_objects[i].object_kind == 1) {
+          red_buoys_array.push_back(task_objects_array->task_objects[i]);
+        }
+        if (task_objects_array->task_objects[i].object_kind == 2) {
+          green_buoys_array.push_back(task_objects_array->task_objects[i]);
+        }
+      }
+
+      RCLCPP_INFO(get_logger(), "red_buoys_array_size : [%ld]", red_buoys_array.size());
+      RCLCPP_INFO(get_logger(), "green_buoys_array_size : [%ld]", green_buoys_array.size());
+
+      RCLCPP_INFO(get_logger(), "red_buoy_x_1 : [%f]", red_buoy_x);
+      RCLCPP_INFO(get_logger(), "red_buoy_y_1 : [%f]", red_buoy_y);
+      RCLCPP_INFO(get_logger(), "green_buoy_x_1 : [%f]", green_buoy_x);
+      RCLCPP_INFO(get_logger(), "green_buoy_y_1 : [%f]", green_buoy_y);
+
+      for (size_t i = 0; i < red_buoys_array.size(); i++) {
```

Comment:
You can write for as

for (const auto & object ; task_objects_array->task_objects)